docs/content/ingestion/data-formats.md

```diff
@@ -72,22 +72,22 @@ If you have nested JSON, [Druid can automatically flatten it for you](flatten-js
 
 ### CSV
 
-Since the CSV data cannot contain the column names (no header is allowed), these must be added before that data can be processed:
-
 ```json
   "parseSpec":{
     "format" : "csv",
     "timestampSpec" : {
       "column" : "timestamp"
     },
     "columns" : ["timestamp","page","language","user","unpatrolled","newPage","robot","anonymous","namespace","continent","country","region","city","added","deleted","delta"],
+    "firstRowIsHeader" : "false",
     "dimensionsSpec" : {
       "dimensions" : ["page","language","user","unpatrolled","newPage","robot","anonymous","namespace","continent","country","region","city"]
     }
   }
 ```
 
-The `columns` field must match the columns of your input data in the same order.
+If your file does not have a header as the first line of the file, you must set the `columns` field and ensure that the order of the fields matches the columns of your input data in the same order.
+If your file does have a header, set `firstRowIsHeader` to true, and do not include the `columns` key.
```

Comment:
It needs to be clear that the new flag will only work in the index task, and not in realtime nor Hadoop.